### PR TITLE
Limit the speed of memory release when drop pending events

### PR DESCRIPTION
**Pull request by xianjingfeng**
### What changes were proposed in this pull request?

Limit the speed of memory release when drop pending events

### Why are the changes needed?

OOM will happen if a large number of events need to be dropped, because usedMemory release immediately, but the speed of GC is not fast enough. #164 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Already added

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java [hunk 1]
```diff
@@ -111,26 +114,28 @@ public ShuffleFlushManager(ShuffleServerConf shuffleServerConf, String shuffleSe
...
-              // todo: get sleep interval from configuration
-              Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
-            }
-          } catch (Exception e) {
-            LOG.error(getName() + " happened exception: ", e);
+    pendingEventProcesser = Executors.newSingleThreadScheduledExecutor(
+            ThreadUtils.getThreadFactory("PendingEventProcesser-%d"));
+    long sleepInterval = shuffleServerConf.get(ShuffleServerConf.PENDING_EVENT_CHECK_INTERVAL);
+    long memoryCapacity = shuffleServerConf.getSizeAsBytes(ShuffleServerConf.SERVER_BUFFER_CAPACITY);;
+    long memoryThreshold = Math.max(memoryCapacity / 10, 1024L * 1024 * 1024);
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:121]: Why do we divide by 10?

**Comment by jerqi** [comment]
A little confused about the effect of pr. Do you give some data on your production environment?

**Comment by jerqi** [comment]
https://stackoverflow.com/questions/8719071/is-it-possible-to-get-outofmemoryerror-because-garbage-collection-too-slow

**Comment by xianjingfeng** [comment]
Discuss in #164?

File: server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java (hunk 1 above)
Comment (xianjingfeng) [reply server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:121]: Use 10 seconds to release all buffer, Maybe bigger than 10 is better, I am observing in our production environment

**Comment by jerqi** [comment]
> Discuss in #164?

OK.

**Pull request closed without merging**